lib/request_handler/schema_parser.rb

```diff
@@ -17,22 +17,22 @@ def initialize(schema:, schema_options: {})
 
     def validate_schema(data, with: schema)
       raise MissingArgumentError, data: 'is missing' if data.nil?
-      data = deep_symbolize_keys(data) if with.rules.keys.first.is_a?(Symbol)
+      if RequestHandler.backend == Validation::DryBackend
+        data = deep_symbolize_keys(data) if with.rules.keys.first.is_a?(Symbol)
+      else
+        data = deep_symbolize_keys(data)
+      end
       validator = validate(data, schema: with)
       validation_failure?(validator)
       validator.output
     end
 
     def validate(data, schema:)
-      if schema_options.empty?
-        schema.call(data)
-      else
-        schema.with(schema_options).call(data)
-      end
+      RequestHandler.backend.new(data, schema, options: schema_options).tap(&:validate)
```

Comment:
`new.tap(&:validate)` seems a bit weird. How about validate returns a validation result instead of storing the result in the backend instance

Reply:
hmm you're right that is weird. not sure why I did that, it should already return the result, will change